packages/swi_prolog.rb

```diff
@@ -0,0 +1,52 @@
+require 'package'
+
+class Swi_prolog < Package
+  description 'SWI-Prolog offers a comprehensive free Prolog environment. Since its start in 1987, SWI-Prolog development has been driven by the needs of real world applications. SWI-Prolog is widely used in research and education as well as commercial applications. Join over a million users who have downloaded SWI-Prolog.'
+  homepage 'https://www.swi-prolog.org/'
+  @_ver = '8.2.4'
+  version @_ver
+  compatibility 'x86_64', 'i686'
+  source_url "https://www.swi-prolog.org/download/stable/src/swipl-#{@_ver}.tar.gz"
+  source_sha256 'f4bcc78437f9080ab089762e9e6afa7071df7f584c14999b92b9a90a4efbd7d8'
+
+  binary_url({
+    i686: 'https://dl.bintray.com/chromebrew/chromebrew/swi_prolog-8.2.4-chromeos-i686.tar.xz',
+  x86_64: 'https://dl.bintray.com/chromebrew/chromebrew/swi_prolog-8.2.4-chromeos-x86_64.tar.xz'
+  })
+  binary_sha256({
+    i686: 'd8985a0a160baca9d6565d4177a43a840450ef4bb96eefae476a51ce27fec063',
+  x86_64: 'cbb431b0bd28fa1c581071aeb1eeda79b70b4ec3c28c2d07b509054e810cb62b'
+  })
+
+  depends_on 'libyaml'
+  depends_on 'libarchive'
+  depends_on 'libdb'
+  depends_on 'jdk8'
```

Comment:
These dependencies will get installed on `aarch64` and `armv7l` even if the package is not compatible.  Eventually, I'd like to add logic in `crew` to avoid this but for now we will need logic as follows:
```
case ARCH
when 'i686', 'x86_64'
  depends_on 'libyaml'
  depends_on 'libarchive'
  depends_on 'libdb'
  depends_on 'jdk8'
end
```

Reply:
Makes sense.